src/core/reference/include/openvino/reference/utils/type_util.hpp

```diff
@@ -19,6 +19,7 @@ namespace ov {
 template <class T>
 constexpr bool is_floating_point() {
     using U = typename std::decay<T>::type;
-    return std::is_floating_point<U>::value || std::is_same<float16, U>::value || std::is_same<bfloat16, U>::value;
+    return std::is_floating_point<U>::value || std::is_same<float16, U>::value || std::is_same<bfloat16, U>::value ||
+           std::is_same<float8_e4m3, U>::value || std::is_same<float8_e5m2, U>::value;
```

Comment:
we also need to add `nf4`

Reply:
Not added intentionally now as `nf4` is type which use less than 8-bits. I think it could cause some issues as this check has been introduced before nf4 and is used in cases where is possible to navigate on buffers using types of N bytes.

Some testing is required for it as there is others nf4 leftovers would be better to check separately. 
Maybe is possible to update this check and make one for all `ov` floats using bytes as base and other where values are stored less than 8-bits.

Reply:
Right now there is no dedicated type for `nf4` and this check cannot be updated by `nf4` until dedicated type will be created.